rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -1296,24 +1294,54 @@ public void onUncaughtException(Throwable e) {
         error(e.getMessage());
     }
 
-    private String parseScriptResult(String json) {
+    private String parseListScriptResult(String json) {
+        JSONValue listScriptResultJsonValue = this.parseJSON(json);
+        JSONObject scriptResultJsonObject = listScriptResultJsonValue.isObject();
+        if (scriptResultJsonObject == null) {
+            StringBuilder scriptResultStr = new StringBuilder();
+            JSONArray scriptResultJsonArray = listScriptResultJsonValue.isArray();
+            JSONValue currentScriptResultJsonValue;
+            JSONObject currentScriptResultJsonObject;
+            for (int i = 0; i < scriptResultJsonArray.size(); i++) {
+                currentScriptResultJsonValue = scriptResultJsonArray.get(i);
+                currentScriptResultJsonObject = currentScriptResultJsonValue.isObject();
+                scriptResultStr.append(parseScriptResult(currentScriptResultJsonObject)).append("<br/>");
+            }
+            return scriptResultStr.toString();
+        } else {
+            return parseScriptResult(scriptResultJsonObject);
+        }
+    }
+
+    private String parseScriptResult(JSONObject scriptResult) {
         StringBuilder scriptResultStr = new StringBuilder();
```

Comment:
a bit too complex to read....is it possible to refactor it a bit to increase readability?

Reply:
Yes, I created a method for each main entity parsed.